test/extensions/filters/http/decompressor/decompressor_filter_test.cc

```diff
@@ -291,79 +290,76 @@ TEST_P(DecompressorFilterTest, ResponseDecompressionDisabled) {
 }
 
 TEST_P(DecompressorFilterTest, NoDecompressionContentEncodingDoesNotMatch) {
-  EXPECT_CALL(*decompressor_factory_, createDecompressor()).Times(0);
+  EXPECT_CALL(*decompressor_factory_, createDecompressor(_)).Times(0);
   Http::TestRequestHeaderMapImpl headers_before_filter{{"content-encoding", "not-matching"},
                                                        {"content-length", "256"}};
   std::unique_ptr<Http::RequestOrResponseHeaderMap> headers_after_filter =
       doHeaders(headers_before_filter, false /* end_stream */);
-  TestUtility::headerMapEqualIgnoreOrder(headers_before_filter, *headers_after_filter);
 
   expectNoDecompression();
 }
 
 TEST_P(DecompressorFilterTest, NoDecompressionContentEncodingNotCurrent) {
-  EXPECT_CALL(*decompressor_factory_, createDecompressor()).Times(0);
+  EXPECT_CALL(*decompressor_factory_, createDecompressor(_)).Times(0);
   // The decompressor's content scheme is not the first value in the comma-delimited list in the
   // Content-Encoding header. Therefore, compression will not occur.
   Http::TestRequestHeaderMapImpl headers_before_filter{{"content-encoding", "gzip,mock"},
                                                        {"content-length", "256"}};
   std::unique_ptr<Http::RequestOrResponseHeaderMap> headers_after_filter =
       doHeaders(headers_before_filter, false /* end_stream */);
-  TestUtility::headerMapEqualIgnoreOrder(headers_before_filter, *headers_after_filter);
 
   expectNoDecompression();
 }
 
 TEST_P(DecompressorFilterTest, NoResponseDecompressionNoTransformPresent) {
-  EXPECT_CALL(*decompressor_factory_, createDecompressor()).Times(0);
+  EXPECT_CALL(*decompressor_factory_, createDecompressor(_)).Times(0);
   Http::TestRequestHeaderMapImpl headers_before_filter{
       {"cache-control", Http::CustomHeaders::get().CacheControlValues.NoTransform},
       {"content-encoding", "mock"},
       {"content-length", "256"}};
   std::unique_ptr<Http::RequestOrResponseHeaderMap> headers_after_filter =
       doHeaders(headers_before_filter, false /* end_stream */);
-  TestUtility::headerMapEqualIgnoreOrder(headers_before_filter, *headers_after_filter);
 
   expectNoDecompression();
 }
 
 TEST_P(DecompressorFilterTest, NoResponseDecompressionNoTransformPresentInList) {
-  EXPECT_CALL(*decompressor_factory_, createDecompressor()).Times(0);
+  EXPECT_CALL(*decompressor_factory_, createDecompressor(_)).Times(0);
   Http::TestRequestHeaderMapImpl headers_before_filter{
       {"cache-control", fmt::format("{}, {}", Http::CustomHeaders::get().CacheControlValues.NoCache,
                                     Http::CustomHeaders::get().CacheControlValues.NoTransform)},
       {"content-encoding", "mock"},
       {"content-length", "256"}};
   std::unique_ptr<Http::RequestOrResponseHeaderMap> headers_after_filter =
       doHeaders(headers_before_filter, false /* end_stream */);
-  TestUtility::headerMapEqualIgnoreOrder(headers_before_filter, *headers_after_filter);
```

Comment:
Just curious. So here, we don't expect `headers_before_filter` to be the same as `*headers_after_filter` since there is no `EXPECT_THAT()` replacement here.